advanced_functionality/tensorflow_bring_your_own/container/cifar10/cifar10.py

```diff
@@ -0,0 +1,209 @@
+        loss=loss,
+        train_op=train_op)
+
+
+def serving_input_fn():
+    inputs = {INPUT_TENSOR_NAME: tf.placeholder(tf.float32, [None, 32, 32, 3])}
+    return tf.estimator.export.ServingInputReceiver(inputs, inputs)
+
+
+def make_batch(data_dir, batch_size=2):
+    dataset = tf.data.TFRecordDataset(data_dir).repeat()
+
+    dataset = dataset.map(parser, num_parallel_calls=batch_size)
+
+    min_queue_examples = int(45000 * 0.4)
+    # Ensure that the capacity is sufficiently large to provide good random
+    # shuffling.
+    dataset = dataset.shuffle(buffer_size=min_queue_examples + 3 * batch_size)
+
+    dataset = dataset.batch(batch_size)
+    iterator = dataset.make_one_shot_iterator()
+
+    image, label = iterator.get_next()
+
+    return image, label
+
+
+def parser(serialized_example):
+    features = tf.parse_single_example(
+        serialized_example,
+        features={
+            'image': tf.FixedLenFeature([], tf.string),
+            'label': tf.FixedLenFeature([], tf.int64),
+        })
+    image = tf.decode_raw(features['image'], tf.uint8)
+    image.set_shape([DEPTH * HEIGHT * WIDTH])
+
+    image = tf.cast(tf.transpose(tf.reshape(image, [DEPTH, HEIGHT, WIDTH]), [1, 2, 0]), tf.float32)
+    label = tf.cast(features['label'], tf.int32)
+
+    return image, label
+
+
+def train_input_fn(data_dir):
+    with tf.device('/cpu:0'):
+        train_data = os.path.join(data_dir, 'train.tfrecords')
+        image_batch, label_batch = make_batch(train_data, BATCH_SIZE)
+        return {INPUT_TENSOR_NAME: image_batch}, label_batch
+
+
+def eval_input_fn(data_dir):
+    with tf.device('/cpu:0'):
+        eval_data = os.path.join(data_dir, 'eval.tfrecords')
+        image_batch, label_batch = make_batch(eval_data, BATCH_SIZE)
+
+        return {INPUT_TENSOR_NAME: image_batch}, label_batch
+
+
+def train(model_dir, data_dir, train_steps):
+    estimator = tf.estimator.Estimator(model_fn=model_fn, model_dir=model_dir)
+
+    temp_input_fn = functools.partial(train_input_fn, data_dir)
+
+    train_spec = tf.estimator.TrainSpec(temp_input_fn, max_steps=train_steps)
+
+    exporter = tf.estimator.LatestExporter('Servo', serving_input_receiver_fn=serving_input_fn)
+    temp_eval_fn = functools.partial(eval_input_fn, data_dir)
+    eval_spec = tf.estimator.EvalSpec(temp_eval_fn, steps=1, exporters=exporter)
+
+    tf.estimator.train_and_evaluate(estimator=estimator, train_spec=train_spec, eval_spec=eval_spec)
+
+
+def main(model_dir, data_dir, train_steps):
+    tf.logging.set_verbosity(tf.logging.INFO)
+    train(model_dir, data_dir, train_steps)
+
+
+if __name__ == '__main__':
+    args_parser = argparse.ArgumentParser()
+    args_parser.add_argument(
+        '--data-dir',
+        default='/opt/ml/input/data/training',
+        type=str,
+        # required=True,
+        help='The directory where the CIFAR-10 input data is stored.')
+    args_parser.add_argument(
+        '--model-dir',
+        default='/opt/ml/model',
```

Comment:
You can do later: explain opt ml model folder

Reply:
Added